modules/Core/src/main/java/org/terasology/core/world/generator/rasterizers/SolidRasterizer.java

```diff
@@ -93,51 +94,53 @@ public void generateChunk(CoreChunk chunk, Region chunkRegion) {
-                    return grass;
-                } else if (depth == 0 && height >= seaLevel + 96) {
-                    return snow;
-                } else if (depth > 32) {
-                    return stone;
-                } else {
-                    return dirt;
-                }
-            case SNOW:
-                if (depth == 0 && height > seaLevel) {
-                    // Snow on top
-                    return snow;
-                } else if (depth > 32) {
-                    // Stone
-                    return stone;
-                } else {
-                    // Dirt
-                    return dirt;
-                }
-            case DESERT:
-                if (depth > 8) {
-                    // Stone
-                    return stone;
-                } else {
-                    return sand;
-                }
-            case OCEAN:
-                if (depth == 0) {
-                    return sand;
-                } else {
-                    return stone;
-                }
-            case BEACH:
-                if (depth < 3) {
-                    return sand;
-                } else {
-                    return stone;
-                }
+    private Block getSurfaceBlock(int depth, int height, Biome type, int seaLevel) {
+        if (type instanceof CoreBiome) {
+            switch ((CoreBiome) type) {
+                case FOREST:
+                case PLAINS:
+                case MOUNTAINS:
+                    // Beach
+                    if (depth == 0 && height > seaLevel && height < seaLevel + 96) {
+                        return grass;
+                    } else if (depth == 0 && height >= seaLevel + 96) {
+                        return snow;
+                    } else if (depth > 32) {
+                        return stone;
+                    } else {
+                        return dirt;
+                    }
+                case SNOW:
+                    if (depth == 0 && height > seaLevel) {
+                        // Snow on top
+                        return snow;
+                    } else if (depth > 32) {
+                        // Stone
+                        return stone;
+                    } else {
+                        // Dirt
+                        return dirt;
+                    }
+                case DESERT:
+                    if (depth > 8) {
+                        // Stone
+                        return stone;
+                    } else {
+                        return sand;
+                    }
+                case OCEAN:
+                    if (depth == 0) {
+                        return sand;
+                    } else {
+                        return stone;
+                    }
+                case BEACH:
+                    if (depth < 3) {
+                        return sand;
+                    } else {
+                        return stone;
+                    }
+            }
         }
         return dirt;
```

Comment:
I did some poking.  This rasterizer is not quite very intelligent in the first place.  This biome to block conversion only happens to the top 32 surface blocks.  So trying to tweak it in an understandable way for world gen extension may be a lost cause because there are no usable biome to block rules that apply to all depths/densities.

At the moment, we could do one of a few things:
1. Do not set the block when there is no mapping to biome.
2. Set the block to air when there is no mapping to biome.
3. Continue setting a fallback block of dirt.

We should probably devise a best practice for how rasterizers should behave so that extending the world gen is understandable.